src/ngAnimate/animateQueue.js

```diff
@@ -0,0 +1,558 @@
+            break;
+        }
+
+        function filterFromRegistry(list, matchContainer, matchCallback) {
+          var containerNode = matchContainer[0];
+          return list.filter(function(entry) {
+            var isMatch = entry.node === containerNode &&
+                            (!matchCallback || entry.callback === matchCallback);
+            return !isMatch;
+          });
+        }
+      },
+
+      push: function(element, event, options, domOperation) {
+        element = stripCommentsFromElement(element);
+        options = options || {};
+        options.domOperation = domOperation;
+        return queueAnimation(element, event, options);
+      },
+
+      enabled: function(bool) {
+        var argCount = arguments.length;
+        if (isElement(bool)) {
+          var element = bool;
+          var node = element.length ? element[0] : element;
+          var recordExists = disabledElementsLookup.get(node);
+
+          // if nothing is set then the animation is enabled
+          bool = !recordExists;
+
+          if (argCount > 1) {
+            bool = !!arguments[1];
+            if (!bool) {
+              disabledElementsLookup.put(node, true);
+            } else if (recordExists) {
+              disabledElementsLookup.remove(node);
+            }
+          }
+        } else if (argCount === 1) {
+          bool = animationsEnabled = !!bool;
+        } else {
+          bool = animationsEnabled;
+        }
+
+        return bool === true;
+      }
+    };
+
+    function queueAnimation(element, event, options) {
+      options = $$animateOptions(element, options);
+      var node = element[0];
+      var parent = element.parent();
+
+      // we create a fake runner with a working promise.
+      // These methods will become available after the digest has passed
+      var runner = new $$AnimateRunner();
+
+      // there are situations where a directive issues an animation for
+      // a jqLite wrapper that contains only comment nodes... If this
+      // happens then there is no way we can perform an animation
+      if (!node) {
+        runner.end();
+        return runner;
+      }
+
+      if (isArray(options.addClass)) {
+        options.addClass = options.addClass.join(' ');
+      }
+
+      if (isArray(options.removeClass)) {
+        options.removeClass = options.removeClass.join(' ');
+      }
+
+      if (options.from && !isObject(options.from)) {
+        options.from = null;
+      }
+
+      if (options.to && !isObject(options.to)) {
+        options.to = null;
+      }
+
+      var className = [node.className, options.addClass, options.removeClass].join(' ');
+      if (!isAnimatableClassName(className)) {
+        runner.end();
+        return runner;
+      }
+
+      var isStructural = ['enter', 'move', 'leave'].indexOf(event) >= 0;
+      var existingAnimation = activeAnimationsLookup.get(node) || {};
+      var existingAnimationExists = !!existingAnimation.state;
```

Comment:
lol it took me a moment to figure out what you meant. `existingAnimationExists` -> `hasExistingAnimation`.